spec/v1/function-builder.spec.ts

```diff
@@ -472,11 +472,59 @@ describe('FunctionBuilder', () => {
     ).to.throw();
   });
 
-  it('', () => {
+  it('should throw an error if private identifier is in the invoker array', () => {
     expect(() =>
       functions.runWith({
         invoker: ['service-account1', 'private', 'service-account2'],
       })
     ).to.throw();
   });
+
+  it('should allow valid secret config expressed using short form', () => {
+    const secrets = ['API_KEY', 'API_KEY@3', 'API_KEY@latest'];
+    const fn = functions
+      .runWith({ secrets })
+      .auth.user()
+      .onCreate((user) => user);
+
+    expect(fn.__trigger.secrets).to.deep.equal(secrets);
+  });
+
+  it('should allow valid secret config expressed with full resource name', () => {
+    const secrets = [
+      'projects/my-project/secrets/API_KEY',
```

Comment:
Can we cut full resource names from scope? I'd like to not create more of a legacy burden than we need? I'm also worried about increasing scope to include cross-project secrets.

Reply:
Discussed offline. Not supporting full resource name (i.e. cross project secrets) in the first iteration.